plugins/base/src/main/resources/dokka/scripts/platform-content-handler.js

```diff
@@ -31,12 +31,16 @@ window.addEventListener('load', () => {
 const darkModeSwitch = () => {
     const localStorageKey = "dokka-dark-mode"
     const storage = localStorage.getItem(localStorageKey)
-    const savedDarkMode = storage ? JSON.parse(storage) : false
+    const osDarkSchemePreferred = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
+    const savedDarkMode = storage ? JSON.parse(storage) : osDarkSchemePreferred
     const element = document.getElementById("theme-toggle-button")
+    const enabledClasses = document.getElementsByTagName("html")[0].classList
     initPlayground(savedDarkMode ? samplesDarkThemeName : samplesLightThemeName)
+    if (savedDarkMode != enabledClasses.contains('theme-dark')) {
+        enabledClasses.toggle("theme-dark");
+    }
```

Comment:
If you have a look at each `.html` page, you'll see this script in the header (inspect [this page](https://kotlinlang.org/api/kotlinx.coroutines/)):
```js
<script>const storage = localStorage.getItem("dokka-dark-mode")
const savedDarkMode = storage ? JSON.parse(storage) : false
if(savedDarkMode === true){
    document.getElementsByTagName("html")[0].classList.add("theme-dark")
}</script>
```
This script is added in [this template](https://github.com/Kotlin/dokka/blob/master/plugins/base/src/main/resources/dokka/templates/base.ftl#L12-L18) with the following comment:
> This script doesn't need to be there but it is nice to have
> since app in dark mode doesn't 'blink' (class is added before it is rendered) 

So I'm thinking maybe it's better to have toggling of themes in one place only, that is in the header script of each page (so add it to the template). This way it's consistent + possibly avoids theme blinking. Something like

```js
const storage = localStorage.getItem("dokka-dark-mode")
if (storage == null) {
    // use OS preferred theme
} else {
    // use theme from storage
}
```

_______

Code inside this file (`platform-content-handler.js`) is mostly adding a click event listener + initizling playground (a.k.a. runnable samples) with the correct theme, so your code is good here. I would just move the class toggling part and leave everything else

Reply:
Thanks for the detailed explanation! I will move the logic for toggling the theme into the template.